karapace/kafka_rest_apis/__init__.py

```diff
@@ -163,16 +167,52 @@ def _add_kafka_rest_routes(self) -> None:
         self.route("/topics/<topic:path>", callback=self.topic_details, method="GET", rest_request=True)
         self.route("/topics/<topic:path>", callback=self.topic_publish, method="POST", rest_request=True)
 
-    async def get_producer(self) -> AIOKafkaProducer:
-        if self.producer_queue.empty():
-            for _ in range(self.config["producer_count"]):
-                self.log.info("Creating async producers")
-                p = await self._create_async_producer()
-                await self.producer_queue.put(p)
-                self.producer_refs.append(p)
-        return await self.producer_queue.get()
+    async def maybe_initilize_producer_queue(self) -> asyncio.Queue:
+        """Lazily initalize Producers."""
+        async with self._producer_queue_lock:
+            if self._producer_queue is None:
+                producer_queue = asyncio.Queue()
+                for _ in range(self.config["producer_count"]):
+                    try:
+                        producer = await self._create_async_producer()
+                    except:
```

Comment:
- even though `_create_async_producer` technically never throws, it is possible for a `KeyboardException` to be raised here, so exceptions should be caught for cleanup.